opensrp-reveal/src/main/java/org/smartregister/reveal/fragment/AvailableOfflineMapsFragment.java

```diff
@@ -0,0 +1,253 @@
+        super.onCreate(savedInstanceState);
+        if (presenter == null) {
+            presenter = new AvailableOfflineMapsPresenter(this);
+        }
+        btnDownloadMap = null;
+
+        new FileHttpServerTask(getContext()).execute();
+
+
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_offline_map, container, false);
+        setUpViews(view);
+        initializeAdapter();
+        return view;
+    }
+
+    private void setUpViews(View view) {
+        offlineMapRecyclerView = view.findViewById(R.id.offline_map_recyclerView);
+
+        btnDownloadMap = view.findViewById(R.id.download_map);
+
+        btnDownloadMap.setOnClickListener(onClickListener);
+
+    }
+
+    private void initializeAdapter() {
+        adapter = new AvailableOfflineMapAdapter(this.getContext(), onClickListener);
+        offlineMapRecyclerView.setAdapter(adapter);
+        if (offlineMapModelList != null) {
+            setOfflineMapModelList(offlineMapModelList);
+        }
+
+    }
+
+    private View.OnClickListener onClickListener = new View.OnClickListener(){
+        @Override
+        public void onClick(View view) {
+            switch (view.getId()) {
+                case R.id.offline_map_checkbox:
+                    updateOperationalAreasToDownload(view);
+                    break;
+                case R.id.download_map:
+                    initiateMapDownload();
+                    break;
+                default:
+                        break;
+            }
+
+        }
+    };
+
+    @Override
+    public void setOfflineMapModelList(List<OfflineMapModel> offlineMapModelList) {
+        if (offlineMapModelList == null) {
+            return;
+        }
+        if (adapter == null) {
+            this.offlineMapModelList = offlineMapModelList;
+        } else {
+            adapter.setOfflineMapModels(offlineMapModelList);
+            this.offlineMapModelList = offlineMapModelList;
+        }
+    }
+
+    public void updateOperationalAreasToDownload(View view) {
+        CheckBox checkBox = (CheckBox) view;
+        OfflineMapModel offlineMapModel = (OfflineMapModel) view.getTag(R.id.offline_map_checkbox);
+
+        if (checkBox.isChecked()) {
+            offlineMapModel.setOfflineMapStatus(OfflineMapModel.OfflineMapStatus.SELECTED_FOR_DOWNLOAD);
+            operationalAreasToDownload.add(offlineMapModel.getLocation());
+        } else {
+            operationalAreasToDownload.remove(offlineMapModel.getLocation());
+        }
+    }
+
+    @Override
+    public void disableCheckBox(String operationalAreaId) {
+        if (adapter ==null) {
+            return;
+        }
+        for (OfflineMapModel offlineMapModel: offlineMapModelList ) {
+            if (offlineMapModel.getDownloadAreaId().equals(operationalAreaId)){
```

Comment:
combine with `(offlineMapModel.getOfflineMapStatus() != OfflineMapModel.OfflineMapStatus.DOWNLOAD_STARTED` ) and remove nested if